Source/FakeItEasy/Priority.cs

```diff
@@ -0,0 +1,169 @@
+﻿namespace FakeItEasy
+{
+    using System;
+
+    /// <summary>
+    /// Indicates precedence between otherwise indistinguishable options, for example when deciding
+    /// which <see cref="IDummyFactory"/> should be used to create a Dummy of a given type.
+    /// In making such decisions, the higher-valued <c>Priority</c> will be used.
+    /// </summary>
+    public struct Priority : IComparable<Priority>, IEquatable<Priority>
+    {
+        private readonly short value;

```

Comment:
We should probably use `int` rather than `short`. See http://stackoverflow.com/a/131354/49241